lib/views/request/request_view.dart

```diff
@@ -0,0 +1,170 @@
+import 'package:clnapp/api/api.dart';
+import 'package:clnapp/views/pay/qr_screen.dart';
+import 'package:flutter/material.dart';
+import '../../components/buttons.dart';
+import '../../model/app_model/generate_invoice.dart';
+import '../../utils/app_provider.dart';
+
+class RequestView extends StatefulWidget {
+  final AppProvider provider;
+
+  const RequestView({Key? key, required this.provider}) : super(key: key);
+
+  @override
+  State<RequestView> createState() => _RequestViewState();
+}
+
+class _RequestViewState extends State<RequestView> {
+  String display = '';
+
+  Future<AppGenerateInvoice> generateInvoice() async {
+    String label = '${DateTime.now()}';
+    String description =
+        "This is a new description created at${DateTime.now()}";
+    AppGenerateInvoice response = await widget.provider
+        .get<AppApi>()
+        .generateInvoice(label, description, amount: int.parse(display));
```

Comment:
```suggestion
        .generateInvoice(label, description, amount: int.parse(display));
```

` int.parse(display)` this is not valid, for floating point values this crash

Reply:
We can for now show the error dialog for the msats like if the value is in decimal then the floor value would be considered.

Reply:
Yeah let do that